qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java

```diff
@@ -30,6 +31,30 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
     private static final String COMPUTERNAME = "WindowsComputername";
     private static final String HOSTNAME = "LinuxDarwinHostname";
 
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .setting("xpack.security.enabled", "false")
+        .withNode(localNodeSpecBuilder -> localNodeSpecBuilder.name(OS_NAME.startsWith("Windows") ? COMPUTERNAME : HOSTNAME))
+        // TODO @jozala -
+        .settingsModifier(
+            settings -> settings.entrySet()
+                .stream()
+                .filter(it -> it.getKey().equals("node.name") == false)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+        )
```

Comment:
I've added this settings modifier in configuration to be able to remove `node.name` from configuration, but keep the rest of the Elasticsearch node setup working. I am wondering if there is already a better way to achieve this.